internal/server.go

```diff
@@ -369,3 +401,32 @@ func (s *Server) getGroupsFromSession(r *http.Request) ([]string, error) {
 	}
 	return groups, nil
 }
+
+func (s *Server) authzIsBypassed(r *http.Request) bool {
+	bypassed := false
+	for _, bypassURI := range config.AuthZPassThrough {
+		if bypassURI == r.URL.Path || (strings.HasSuffix(bypassURI, "*") && strings.HasPrefix(r.URL.Path, strings.TrimRight(bypassURI, "*"))) {
```

Comment:
This URI matching seems like a very important code that is also duplicated in `NonResourceURLMatches`. I'd extract it to own function with proper unit tests.